base/special/exp.jl

```diff
@@ -425,10 +431,25 @@ end
                       0.008332997481506921, 0.0013966479175977883, 0.0002004037059220124))
     small_part = r*hi
     twopk = reinterpret(Float64, (N+1023) << 52)
-    x > MAX_EXP(Float32) && return Inf32
     return Float32(muladd(twopk, small_part, twopk-1.0))
 end
 
+@inline function expm1(x::Float16)
```

Comment:
Unrelated to the core of the PR, but why are we forcing inline'ing with `exmp1`? The Float64 method looks especially large. If I recall correctly, I thought we had a discussion sometime against automatically forcing inline on the math methods to prevent excessively large code gen?

Reply:
TLDR is I don't remember, and it arguably was a mistake. That said, I don't think it's one worth fixing here.

Reply:
Since that PR is merged I updated this PR.